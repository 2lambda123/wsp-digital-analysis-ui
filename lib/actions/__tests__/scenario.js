/* global describe, expect, it */

describe('actions > scenario', () => {
  const nock = require('nock')
  const scenario = require('../scenario')
  const {makeMockStore} = require('../../utils/mock-data')

  it('create should work', async () => {
    nock('http://mockHost.com/').post('/api/scenario').reply(200, {id: '1'})

    const action = scenario.create({
      bundleId: '1',
      name: 'test',
      projectId: '1'
    })
    const store = makeMockStore()
    const actionResult = store.dispatch(action)
    const fetchResult = await actionResult[1]

    delete action.payload.options.body.id
    expect(action).toMatchSnapshot()
    expect(actionResult).toMatchSnapshot()
    expect(fetchResult).toMatchSnapshot()
  })

  it('deleteScenario should work', async () => {
    nock('http://mockHost.com/').delete('/api/scenario/1').reply(200, 'deleted')

    const action = scenario.deleteScenario({
      projectId: '1',
      scenarioId: '1'
    })
    const store = makeMockStore()
    expect(action).toMatchSnapshot()
    const actionResult = store.dispatch(action)
    expect(actionResult).toMatchSnapshot()
    const fetchResult = await actionResult[1]
    expect(fetchResult).toMatchSnapshot()
  })

  it('load should work', async () => {
    nock('http://mockHost.com/')
      .get('/api/scenario/1')
      .reply(200, {id: '1', bundleId: '1'})
      .get('/api/scenario/1/modifications')
      .reply(200)

    const action = scenario.load('1')
    const store = makeMockStore()
    expect(action).toMatchSnapshot()
    const actionResult = store.dispatch(action)
    expect(actionResult).toMatchSnapshot()
    const fetchResult = await actionResult[1]
    expect(fetchResult).toMatchSnapshot()
  })

  it('saveToServer should work', async () => {
    nock('http://mockHost.com/').put('/api/scenario/1').reply(200, {id: '1'})

    const action = scenario.saveToServer({id: '1'})
    const store = makeMockStore()
    expect(action).toMatchSnapshot()
    const actionResult = store.dispatch(action)
    expect(actionResult).toMatchSnapshot()
    const fetchResult = await actionResult[1]
    expect(fetchResult).toMatchSnapshot()
  })

  it('set should work', () => {
    expect(scenario.set()).toMatchSnapshot()
  })

  it('setAll should work', () => {
    expect(scenario.setAll()).toMatchSnapshot()
  })

  it('setAndLoadModifications should work', async () => {
    nock('http://mockHost.com/').get('/api/scenario/1/modifications').reply(200)

    const action = scenario.setAndLoadModifications({id: '1', bundleId: '1'})
    const store = makeMockStore()
    const actionResult = store.dispatch(action)
    const fetchResult = await actionResult[1]

    expect(action).toMatchSnapshot()
    expect(actionResult).toMatchSnapshot()
    expect(fetchResult).toMatchSnapshot()
  })

  it('showVariant should work', () => {
    expect(scenario.showVariant()).toMatchSnapshot()
  })

  it('updateVariant should work', () => {
    expect(scenario.updateVariant()).toMatchSnapshot()
  })

  it('updateVariants should work', () => {
    expect(scenario.updateVariants()).toMatchSnapshot()
  })
})
