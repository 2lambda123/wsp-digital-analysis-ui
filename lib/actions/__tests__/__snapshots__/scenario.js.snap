exports[`actions > scenario create should work 1`] = `
Object {
  "type": "increment outstanding requests",
}
`;

exports[`actions > scenario create should work 2`] = `
Object {
  "type": "decrement outstanding requests",
}
`;

exports[`actions > scenario create should work 3`] = `
Array [
  Object {
    "payload": Object {
      "id": "1",
    },
    "type": "add scenario",
  },
  Object {
    "payload": Object {
      "args": Array [
        "/projects/1/scenarios/1",
      ],
      "method": "push",
    },
    "type": "@@router/CALL_HISTORY_METHOD",
  },
]
`;

exports[`actions > scenario createVariant should work 1`] = `
Object {
  "type": "create variant",
}
`;

exports[`actions > scenario deleteScenario should work 1`] = `
Object {
  "type": "increment outstanding requests",
}
`;

exports[`actions > scenario deleteScenario should work 2`] = `
Object {
  "type": "decrement outstanding requests",
}
`;

exports[`actions > scenario deleteScenario should work 3`] = `
Array [
  Object {
    "payload": "1",
    "type": "delete scenario",
  },
  Object {
    "payload": Object {
      "args": Array [
        "/projects/1",
      ],
      "method": "push",
    },
    "type": "@@router/CALL_HISTORY_METHOD",
  },
]
`;

exports[`actions > scenario load should work 1`] = `
Object {
  "type": "increment outstanding requests",
}
`;

exports[`actions > scenario load should work 2`] = `
Object {
  "type": "decrement outstanding requests",
}
`;

exports[`actions > scenario load should work 3`] = `
Array [
  Object {
    "payload": Object {
      "bundleId": "1",
      "id": "1",
    },
    "type": "set scenario",
  },
  Array [
    Object {
      "type": "increment outstanding requests",
    },
    Promise {},
  ],
]
`;

exports[`actions > scenario saveToServer should work 1`] = `
Object {
  "type": "increment outstanding requests",
}
`;

exports[`actions > scenario saveToServer should work 2`] = `
Object {
  "type": "decrement outstanding requests",
}
`;

exports[`actions > scenario saveToServer should work 3`] = `
Object {
  "payload": Object {
    "id": "1",
  },
  "type": "set scenario",
}
`;

exports[`actions > scenario set should work 1`] = `
Object {
  "type": "set scenario",
}
`;

exports[`actions > scenario setAll should work 1`] = `
Object {
  "type": "set scenarios",
}
`;

exports[`actions > scenario setAndLoadModifications should work 1`] = `
Array [
  Object {
    "payload": Object {
      "bundleId": "1",
      "id": "1",
    },
    "type": "set scenario",
  },
  Array [
    Object {
      "type": "increment outstanding requests",
    },
    Promise {},
  ],
]
`;

exports[`actions > scenario showVariant should work 1`] = `
Object {
  "type": "show variant",
}
`;

exports[`actions > scenario updateVariant should work 1`] = `
Object {
  "type": "update variant",
}
`;

exports[`actions > scenario updateVariants should work 1`] = `
Object {
  "type": "update variants",
}
`;
